utils/src/main/java/com/cloud/utils/net/NetUtils.java

```diff
@@ -268,19 +268,7 @@ public static String getDefaultEthDevice() {
             final String defDev = Script.runSimpleBashScript("/sbin/route -n get default 2> /dev/null | grep interface | awk '{print $2}'");
             return defDev;
         }
-        final String defaultRoute = Script.runSimpleBashScript("/sbin/route | grep default");
-
-        if (defaultRoute == null) {
-            return null;
-        }
-
-        final String[] defaultRouteList = defaultRoute.split("\\s+");
-
-        if (defaultRouteList.length != 8) {
-            return null;
-        }
-
-        return defaultRouteList[7];
+        return Script.runSimpleBashScript("awk '$2 == 00000000 { print $1 }' /proc/net/route");
```

Comment:
in @wido 's style i'd prefer: (whatever happened to "suggest change"?)
```ip route show default 0.0.0.0/0|head -1|awk '{print $5}'```
note that we do need the fifth field, i.e. return the device not an address.